packages/webarchitect/src/validateFileRule.ts

```diff
@@ -0,0 +1,176 @@
+
+/**
+ * Generates a human-readable description of what a JSON schema validates.
+ * Summarizes key constraints like type, required properties, patterns, etc.
+ * Used to provide clear feedback about what validation rules are being applied.
+ *
+ * @param schema - JSON Schema to describe
+ * @returns Human-readable description of the schema's validation rules
+ *
+ * @example
+ * ```typescript
+ * const schema = {
+ *   type: "object",
+ *   required: ["name", "version"],
+ *   properties: { name: {}, version: {} }
+ * };
+ * const desc = describeSchema(schema);
+ * // Returns: "must be object, must have properties: name, version, expected properties: name, version"
+ * ```
+ */
+function describeSchema(schema: JSONSchema7): string {
+  const descriptions: string[] = [];
+
+  if (schema.type) {
+    if (typeof schema.type === "string") {
+      descriptions.push(`must be ${schema.type}`);
+    } else if (Array.isArray(schema.type)) {
+      descriptions.push(`must be one of: ${schema.type.join(", ")}`);
+    }
+  }
+
+  if (schema.const !== undefined) {
+    descriptions.push(`must equal "${schema.const}"`);
+  }
+
+  if (typeof schema.pattern === "string") {
+    descriptions.push(`must match pattern /${schema.pattern}/`);
+  }
+
+  if (schema.required && Array.isArray(schema.required)) {
+    descriptions.push(`must have properties: ${schema.required.join(", ")}`);
+  }
+
+  if (schema.properties && typeof schema.properties === "object") {
+    const propNames = Object.keys(schema.properties);
+    if (propNames.length <= 3) {
+      descriptions.push(`expected properties: ${propNames.join(", ")}`);
+    } else {
+      descriptions.push(`validates ${propNames.length} properties`);
+    }
+  }
+
+  if (descriptions.length === 0) {
+    return "validates file content structure";
+  }
+
+  return descriptions.join(", ");
+}
+
+/**
+ * Validates a file against a file rule specification.
+ * Reads the file, parses it as JSON, and validates against the provided schema.
+ * Provides detailed error messages for missing files, invalid JSON, or schema violations.
+ *
+ * @param projectPath - Base path where the file should exist
+ * @param fileRule - File rule specification including name and JSON schema
+ * @param ruleName - Name of the rule for error reporting (e.g., "package-config")
+ * @returns Promise that resolves to an array with a single validation result
+ *
+ * @example
+ * ```typescript
+ * const rule = {
+ *   name: "package.json",
+ *   contains: {
+ *     type: "object",
+ *     properties: {
+ *       name: { type: "string" },
+ *       version: { type: "string" }
+ *     },
+ *     required: ["name", "version"]
+ *   }
+ * };
+ * const results = await validateFileRule("/project", rule, "package-config");
+ * ```
+ */
+export default async function validateFileRule(
+  projectPath: string,
+  fileRule: FileRule,
+  ruleName: string, // Add rule name parameter
+): Promise<ValidationResult[]> {
```

Comment:
This returns an array of validation results, but you only ever return an array of one result, and each error case short-circuit returns an error result. 

Maybe you should construct an array `results` and push each result onto it, so that all errors are returned, rather than just whichever one came first in your validation